pandas/io/pytables.py

```diff
@@ -266,7 +266,7 @@ def to_hdf(
     format: str | None = None,
     index: bool = True,
     min_itemsize: int | dict[str, int] | None = None,
-    nan_rep=None,
+    nan_rep: Any | None | None = None,
```

Comment:
I"m not sure this is correct.  It might be that the docs are incorrect here in saying `Any`, but I think it needs to be `str | None`.  Also, you have an extra `None` here.

If you make the change here, you should also make it in `pandas/core/generic.py` as well